src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -42,7 +49,18 @@ internal ItemManager(RecentItemsManager recentItemsManager)
             IAsyncCompletionSession session,
             AsyncCompletionSessionInitialDataSnapshot data,
             CancellationToken cancellationToken)
-            => Task.FromResult(data.InitialList.OrderBy(i => i.SortText).ToImmutableArray());
+        {
+            if (session.TextView.Properties.TryGetProperty(CompletionSource.TargetTypeFilterExperimentEnabled, out bool isExperimentEnabled) && isExperimentEnabled)
+            {
+                // This method is called exactly once, so use the opportunity to set a baseline for telemetry.
+                if (data.InitialList.Any(i => i.Filters.Any(f => f.DisplayText == FeaturesResources.Target_type_matches)))
+                {
+                    Logger.Log(FunctionId.Intellisense_AsyncCompletion_SessionContainsTargetTypeFilter);
```

Comment:
@heejaechang Do I need to be doing something stronger than this? I see these calls being hit but nothing in telemetryexplorer.